state/execution.go

```diff
@@ -916,6 +917,25 @@ func getLogs(responses []*abci.ExecTxResult) []string {
 	return logs
 }
 
+// FindAvailableFilename finds an available filename by adding suffix if file already exists
+func FindAvailableFilename(dir, baseFilename string) string {
+	filename := baseFilename
+	suffix := 0
+	for {
```

Comment:
make this for loop stop after a certain number of iterations, like 100, and return an error. Just as a fool proof mechanism